### DISCUSSION TITLE
Populate a repeater on a create form with relations (within a relations manager)

### DISCUSSION THREAD

**Discussion by StaffCollab** [filamentphp/filament#6920] [Help] [answered]
Is it possible to populate a repeater on a create form to include all advertisements and charges associated with a payment? 

Currently, I have an account page with a relations manager for payments. When a new payment is created, I want to automatically add all associated advertisements and charges. 

However, the repeater remains empty because the account field has not yet been saved in the database. Are there any workarounds to achieve this?

**Comment by danharrin** [comment]
I can't think of a way at the moment, it doesn't sound like a very common situation. Maybe try using `default()` on the repeater?

**Reply by StaffCollab** [reply]
Can we use the default() on the repeater itself and then auto-generate the children?

It doesn't seem that this functionality is available. Pls clarify

**Reply by danharrin** [reply, marked as answer]
Yeah, `Repeater::make()->default([['name' => 'x'], ['name' => 'y']])`
